### ISSUE TITLE
@tailwindcss/typography: ul, li, ol have "list-style: none" after upgrading from 3.0.0 to 3.0.4

### ISSUE THREAD

**Issue by TommasoAmici** [tailwindlabs/tailwindcss#6527]
**What version of Tailwind CSS are you using?**

v3.0.4

**What build tool (or framework if it abstracts the build tool) are you using?**

```json
"autoprefixer": "^10.4.0",
"postcss": "^8.4.4",
"postcss-loader": "^6.2.1",
"webpack": "^5.65.0",
```

**What version of Node.js are you using?**

For example: v16.13.1

**What browser are you using?**

N/A

**What operating system are you using?**

For example: macOS

**Reproduction URL**

Tailwind Play doesn't have 3.0.4 yet, but this stops working with 3.0.4: https://play.tailwindcss.com/XLagurIOT4

**Describe your issue**

With `@tailwindcss/typography` lists are unstyled when using v3.0.4

**Comment by mwnciau** [comment]
From the [documentation](https://tailwindcss.com/docs/preflight#lists-are-unstyled): "Ordered and unordered lists are unstyled by default, with no bullets/numbers and no margin or padding." So this is intended behaviour.

However, and more importantly, lists are completely broken with the change 00f60e6:

https://play.tailwindcss.com/REkq4bvDIP

This works in v3.0.1, but not in v3.0.4 because the list-style for each li item is getting reset, so the classes on the ul are ignored.

**Comment by RobinMalfait** [comment]
Hey! Thank you for your bug report!
Much appreciated! 🙏

We accidentally broke this, will be fixed in v3.0.5 which is being published right now.
